src/adapter/etl-adapter-csv/src/Flow/ETL/Adapter/CSV/CSVExtractor.php

```diff
@@ -184,4 +202,26 @@ private function mapHeaders(array $headers) : array
             \array_keys($headers)
         );
     }
+
+    /**
+     * Detects UTF-8 BOM at the beginning of the stream and returns the offset to skip it.
+     * 
+     * @return int The number of bytes to skip (3 if BOM is present, 0 otherwise)
+     */
+    private function detectBOM(SourceStream $stream) : int
```

Comment:
This method is just broken. It introduces unnecessary complexity. Besides reading the stream twice (which creates unnecessary HTTP requests for remote streams), it returns int. 
You might ask yourself why int? Well, it's an offset to skip when BOM is detected. 
But here is the twist: BOM can be only 3 bytes... So AI simply increased the complexity of our code.

Instead of returning the offset, that method (still unnecessary) should return bool to tell us if the given line starts from BOM or not.